qpid-jms-client/src/main/java/org/apache/qpid/jms/transports/netty/NettyTcpTransport.java

```diff
@@ -233,15 +228,13 @@ public void close() throws IOException {
         if (closed.compareAndSet(false, true)) {
             connected.set(false);
             try {
-                if (channel != null) {
+                if (channel != null && groupRef != null) {
```

Comment:
It didnt see like this should be changing behaviour in a noticable way, as the group [ref] isnt being created or closed at a different point than before, so it just seems odd this check of groupRef would be needed here when this bit doesnt actually close/release the groupRef.

But actually, thinking on it more, I guess it really _does_ change the group behaviour; the shared executor will _continue_ running after the provider thinks it closed it down, if other connections are still using it. Thus with these changes it could continue executing things for the provider/connection even after we currently expect it cant. That may need further changes to address.

Reply:
I'm investigating in this last bit